File: front/ycyw/src/app/dto/dialog.dto.ts
import { ChatMessageDTO } from './chat-message.dto';
import { UserProfileDTO } from './user-profile.dto';

export type DialogStatus = 'OPEN' | 'PENDING' | 'CLOSED';

export interface DialogDTO {
  id: number;
  topic: string;
  status: DialogStatus;
  createdAt: string;
  closedAt: string;
  lastActivityAt: string;
  participants: UserProfileDTO[];
  messages: ChatMessageDTO[];
}

File: back/poc_chat/src/main/java/com/ycyw/poc_chat/dto/DialogDTO.java
package com.ycyw.poc_chat.dto;

import com.ycyw.poc_chat.model.DialogStatus;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;
import lombok.Data;

@Data
public class DialogDTO {

  private Long id;
  private String topic;
  private DialogStatus status;
  private LocalDateTime createdAt;
  private LocalDateTime closedAt;
  private LocalDateTime lastActivityAt;
  private Set<UserProfileDTO> participants = new HashSet<>();
  private Set<ChatMessageDTO> messages = new HashSet<>();
}
